core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -190,13 +190,13 @@ private class ClientEndpoint(
                 logDebug(s"State of driver $submittedDriverID is ${state.get}, " +
                   s"continue monitoring driver status.")
               }
-            }
-        }
-      } else {
+          }
+      }
+    } else if (exception.isEmpty || !Utils.responseFromBackup(a.get.getMessage())) {
```

Comment:
`a.get`? typo?

Reply:
I am new in Scala. The exception is an Option[Exception]. As I know we can't call getMessage() directly! Am I right?
e.g. https://scastie.scala-lang.org/Zs5zrPU1SqulmCRthPHDXg

Reply:
I mean, I didn't see `a` is defined in the scope. It should be `exception.get.getMessage`.

Reply:
I also changed the condition handling to keep compatible with previous assumptions.